File: competitive_programming/leetcode/01_python/412_fizz_buzz.py
"""LeetCode 412 - Fizz Buzz."""
def fizz_buzz(n):
    result = []
    for value in range(1, n + 1):
        if value % 15 == 0:
            result.append("FizzBuzz")
        elif value % 3 == 0:
            result.append("Fizz")
        elif value % 5 == 0:
            result.append("Buzz")
        else:
            result.append(str(value))
    return result


if __name__ == "__main__":
    assert fizz_buzz(5) == ["1", "2", "Fizz", "4", "Buzz"]
    assert fizz_buzz(15)[-1] == "FizzBuzz"
    print("412 fizz buzz ok")

File: competitive_programming/leetcode/03_cpp/412_fizz_buzz.cpp
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdint>
#include <iostream>
#include <set>
#include <string>
#include <vector>

std::vector<std::string> fizzBuzz(int n) {
    std::vector<std::string> result;
    for (int value = 1; value <= n; ++value) {
        if (value % 15 == 0) result.push_back("FizzBuzz");
        else if (value % 3 == 0) result.push_back("Fizz");
        else if (value % 5 == 0) result.push_back("Buzz");
        else result.push_back(std::to_string(value));
    }
    return result;
}

int main() {
    assert((fizzBuzz(5) == std::vector<std::string>{"1", "2", "Fizz", "4", "Buzz"}));
    assert(fizzBuzz(15).back() == "FizzBuzz");
    std::cout << "412 fizz buzz ok\n";
    return 0;
}
